orca/service_test.go

```diff
@@ -93,16 +63,42 @@ func (s) TestE2E_CustomBackendMetrics_OutOfBand(t *testing.T) {
 	opts := orca.ServiceOptions{MinReportingInterval: shortReportingInterval, ServerMetricsProvider: smr}
 	internal.AllowAnyMinReportingInterval.(func(*orca.ServiceOptions))(&opts)
 
+	var requests int64
+	var mu sync.Mutex
+
+	const numRequests = 20
+
+	stub := &stubserver.StubServer{
+		UnaryCallF: func(ctx context.Context, req *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
+			mu.Lock()
+			requests++
+
+			smr.SetNamedUtilization(requestsMetricKey, float64(requests)*0.01)
```

Comment:
seems like we have changed how the lock is handled here, but this seems to be the right way of doing it. 

But this would go away if we use an atominc.Int64 here.